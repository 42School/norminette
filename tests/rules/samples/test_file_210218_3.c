#define LENGTH 8
typedef char	t_word[8];
typedef char	t_word[LENGTH];

typedef void	(*t_func_definiton)(toto arg);