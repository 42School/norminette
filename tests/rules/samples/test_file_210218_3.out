[36mtest_file_210218_3.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 1":
		<DEFINE=#define LENGTH 8> <NEWLINE>
[36mtest_file_210218_3.c[0m - [32mIsUserDefinedType[0m In "GlobalScope" from "None" line 2":
		<TYPEDEF> <SPACE> <CHAR> <TAB> <IDENTIFIER=t_word> <LBRACKET> <CONSTANT=8> <RBRACKET> <SEMI_COLON> <NEWLINE>
[36mtest_file_210218_3.c[0m - [32mIsUserDefinedType[0m In "GlobalScope" from "None" line 3":
		<TYPEDEF> <SPACE> <CHAR> <TAB> <IDENTIFIER=t_word> <LBRACKET> <IDENTIFIER=LENGTH> <RBRACKET> <SEMI_COLON> <NEWLINE>
[36mtest_file_210218_3.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 4":
		<NEWLINE>
[36mtest_file_210218_3.c[0m - [32mIsUserDefinedType[0m In "GlobalScope" from "None" line 5":
		<TYPEDEF> <SPACE> <VOID> <TAB> <LPARENTHESIS> <MULT> <IDENTIFIER=t_func_definiton> <RPARENTHESIS> <LPARENTHESIS> <IDENTIFIER=toto> <SPACE> <IDENTIFIER=arg> <RPARENTHESIS> <SEMI_COLON> 
test_file_210218_3.c: Error!
Error: INVALID_HEADER       (line:   1, col:   1):	Missing or invalid 42 header
Error: NL_AFTER_PREPROC     (line:   2, col:   1):	Preprocessor statement must be followed by a newline
Error: FORBIDDEN_CHAR_NAME  (line:   3, col:  24):	user defined identifiers should contain only lowercase characters, digits or '_'
